net/if_indextoname.c

```diff
@@ -0,0 +1,42 @@
+/*
+ * Phoenix-RTOS
+ *
+ * libphoenix
+ *
+ * net/if_indextoname.c
+ *
+ * Copyright 2021 Phoenix Systems
+ * Author: Maciej Purski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <unistd.h>
+#include <string.h>
+#include <sys/socket.h>
+#include <sys/ioctl.h>
+#include <net/if.h>
+
+char *if_indextoname(unsigned int ifindex, char *ifname)
+{
+	struct ifreq ifr = {0};
+	int fd, res;
+
+	fd = socket(AF_INET, SOCK_DGRAM, 0);
+	if (fd <= 0)
+		return 0;
+
+	ifr.ifr_ifindex = ifindex;
+	res = ioctl(fd, SIOCGIFNAME, &ifr);
+
+	close(fd);
+	if (res < 0) {
+		return NULL;
+	}
+	else {
+		strncpy(ifname, ifr.ifr_name, IFNAMSIZ);
```

Comment:
```suggestion
		strncpy(ifname, ifr.ifr_name, IFNAMSIZ);
		ifbame[IFNAMSIZ - 1] = '\0';
```
eliminating buffer overflows

Reply:
or use `snprintf`, but that's wasteful in this particular case